### PR TITLE
Minor enhancements and crash fixes for i3wm

### PR DESCRIPTION
**Pull request by anhnamtran**
- Handle `XDG_CONFIG_HOME` on systems where `XDG_CONFIG_HOME` is not set. This defaults to `$HOME/.config` [per the XDG Base Directories specifications](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables)
- Automatically create parent directories for cache if the parent directories do not exist
- When not focused on anything, output `Desktop` instead of `Empty`
- When no icon is found or the icon is empty, do not print the gap
- If a config file already exists, running `./install` will no longer overwrite it
- Fix a crash when hot-plugging monitors (`OutputEvent`) or switching binding modes (`ModeEvent`) in i3wm

### CODE REVIEW

**Comment by andreykaere** [comment]
Hi! Please, format your commit messages, according to [this](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) template. More to that, you should open PR to `dev` branch, because I do all the work in that branch and then  merge `dev` to `master`, when it's stable enough.

I want `ixwindow` module to be the extension of `xwindow`, so for compatibility, I chose "Empty". If you want to have the option to put "Desktop" there, then you should make it as an option that can be specified in the config file.

I don't know how I can test the panicking on i3, because I don't have an external monitor, so I am gonna trust you on this one ... Because it shouldn't hurt anyway ...

**Comment by anhnamtran** [comment]
Sure thing. I didn't realize that you had a preference. It's probably helpful to have either a PR template or a little blurb for contributions as it'll probably reduce your workload a bit more.

I will make it so "Empty" is configurable.

For panicking on i3, for the `ModeEvent`, if you have a `Resize` binding mode (mine and on the default config it's `Mod+R`), that would cause the panic. 

For the `OutputEvent`, I just did `xrandr --output <MyOutput> --off` but you would need 2 monitors for that.

**Comment by anhnamtran** [comment]
I see that you've addressed the empty `XDG_CONFIG_HOME` issue in `dev` as well. I will rebase onto that and see which commits are no longer needed. 

As a side note, I see you're pretty actively working on this project. Would it be more convenient for you that I don't make PRs at all and just have a separate fork?

**Comment by andreykaere** [comment]
No, of course your PR is very helpful! I've been meaning to merge some of the critical fixes to the project when I have time anyway. But if you want some features I don't prefer, then you can either make them additional and I will be happy to merge them or just feel free to keep it in your own fork :)

**Comment by andreykaere** [comment]
About lacking CONTRIBUTION.md -- yes, I want to add this, but I can't find time to finally sit and rewrite a major part of code anyway. Besides, I've written quite a few bugs to fix. So, I will definitely do that in the future, just haven't gotten time yet.

Besides there are not so many PRs at the moment, so that's fine :)

**Comment by anhnamtran** [comment]
Okay great! I'll do some git magic to figure out which changes I have that you've potentially handled in the other branches, remove the commits, and reference those commits instead.

**Comment by andreykaere** [comment]
`print_info_feature` is the last branch that I was committing to, I believe. But I don't think there are a lot of duplications in your commits.

**Comment by andreykaere** [comment]
I think it's better to rebase onto `dev`. Maybe it's easier if you copy `dev` to your fork and rebase there and open new PR?

File: src/core.rs [hunk 1]
```diff
@@ -282,7 +282,7 @@ where
         let state = &self.monitor.state;
 
         if state.curr_icon_name.is_none() {
-            println!("Empty");
+            println!("Desktop");
```
Comment (anhnamtran) [inline src/core.rs:285]: Per the comments, make this configurable instead of hard coded.

File: src/core.rs [hunk 2]
```diff
@@ -302,11 +302,14 @@ where
...
-        print!("{}", self.config.gap());
+        print!("{}", match state.curr_icon_name.as_ref().unwrap() {
+            IconName::Empty => "",
+            IconName::Name(_) => self.config.gap(),
+        });
         io::stdout().flush().unwrap();
 
         match state.curr_icon_name.as_ref().unwrap() {
-            IconName::Empty => println!("Empty"),
+            IconName::Empty => println!("Desktop"),
```
Comment (anhnamtran) [inline src/core.rs:315]: Same as above

**Review by anhnamtran — COMMENTED** [review]
Making specific review comments to better track progress of this PR.

**Comment by andreykaere** [comment]
Why don't you want to rebase your changes onto `dev`? `master` is behind `dev` anyway and you will have to rebase (not merge!) onto `dev`anyway for me to accept this PR

**Comment by anhnamtran** [comment]
Hi, I did rebase onto dev. Perhaps the confusion comes from the branch naming of my fork. The fork's `master` branch is now exclusively tracking the `dev` branch.

**Comment by andreykaere** [comment]
Oh, yes, I see it now. Okay, then just request my review once you think thid PR is ready to be merged and I'll take a closer look

**Comment by andreykaere** [comment]
I am going to close this PR due to inactivity. Some of the addressed problems were fixed in the recent merge (when implementing `print_info` feature), consider checking it out. Some of the suggestions I am going to implement manually, thank you for drawing my attention to them!

**Comment by anhnamtran** [comment]
Hey. Sorry for the inactivity and thank you for keeping an eye on this regardless, life got a bit hectic.

**Comment by andreykaere** [comment]
Hi! No problem! In fact, thank you for drawing my attention to some bugs and suggesting fixes!

**Pull request closed without merging**